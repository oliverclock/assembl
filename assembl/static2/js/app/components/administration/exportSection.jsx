// @flow
import React from 'react';
import { connect } from 'react-redux';
import { I18n, Translate } from 'react-redux-i18n';
import { Link } from 'react-router';
import { compose, graphql } from 'react-apollo';
import { FormGroup, Radio, FormControl } from 'react-bootstrap';
import withLoadingIndicator from '../../components/common/withLoadingIndicator';


import { getDiscussionId } from '../../utils/globalFunctions';
import SectionTitle from './sectionTitle';
import DiscussionPreferenceLanguage from '../../graphql/DiscussionPreferenceLanguage.graphql';

type Props = {
  languages?: Array<Object>,
  exportType: string,
  voteSessionId?: string
};

type State = {
  exportLocale: ?string,
  translate: boolean
};

class DumbExportSection extends React.Component<Object, Props, State> {
  props: Props;

  state: State;

  static defaultProps = {
    languages: null,
    voteSessionId: null
  };


  static ExportLanguageDropDown = ({ languages, onSelect, activeKey }) => {
    const activeLanguage = languages.filter(language => language.locale === activeKey)[0];
    return (
      <FormControl
        className="export-language-dropdown"
        componentClass="select"
        onChange={({ target: { value } }) => {
          onSelect(value);
        }}
        value={activeLanguage.locale}
      >
        {languages.map(lang => (
          <option key={`locale-${lang.locale}`} value={lang.locale}>
            {lang.name}
          </option>
        ))}
      </FormControl>
    );
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      exportLocale: null,
      translate: false
    };
  }


  selectExportLocale = (locale: string) => {
    this.setState({ exportLocale: locale });
  };

  toggleTranslation = (shouldTranslate: boolean) => {
    this.setState({ translate: shouldTranslate });
  };

  render() {
    const { languages, exportType, voteSessionId } = this.props;
    const isSurveyExport = exportType === 'survey';
    const { translate } = this.state;
    const debateId = getDiscussionId();
    if (!debateId) return null;
    const exportLocale = this.state.exportLocale || languages && languages[0].locale;
    let exportLink;
    if (isSurveyExport) { exportLink = `/data/Discussion/${debateId}/phase1_csv_export${translate ? `?lang=${exportLocale}` : ''}`; } else {
      exportLink = `/data/Discussion/${debateId}/widgets/${voteSessionId}/vote_results_csv`;
    }
    return (
      <div className="admin-box survey-admin-export-section">
        <SectionTitle title={I18n.t('administration.survey.2')} annotation={I18n.t('administration.surveyExport.annotation')} />
        <div className="admin-content">
          {isSurveyExport && <FormGroup>
            <Radio
              checked={!translate}
              onChange={() => {
                this.toggleTranslation(false);
              }}
            >
              <Translate value="administration.surveyExport.noExportLanguage" />
            </Radio>
            <Radio
              checked={translate}
              onChange={() => {
                this.toggleTranslation(true);
              }}
            >
              <Translate value="administration.surveyExport.translateTheMessagesIn" />
              {translate && (
                <DumbExportSection.ExportLanguageDropDown
                  languages={languages}
                  onSelect={this.selectExportLocale}
                  activeKey={exportLocale}
                />
              )}
            </Radio>
          </FormGroup>}

          <br />
          <Link className="button-link button-dark margin-l" href={exportLink}>
            <Translate value="administration.surveyExport.link" />
          </Link>
        </div>
      </div>
    );
  }
}

export { DumbExportSection };

const mapStateToProps = ({ i18n, admin }) => ({
  i18n: i18n,
  voteSessionId: admin.voteSession.page.toJS().id
});

export default compose(
  connect(mapStateToProps),
  graphql(DiscussionPreferenceLanguage, {
    options: ({ i18n: { locale } }) => ({
      variables: {
        inLocale: locale
      }
    }),
    props: ({ data }) => {
      if (data.loading) {
        return {
          loading: true
        };
      }

      if (data.error) {
        return {
          hasErrors: true,
          loading: false
        };
      }
      return {
        hasErrors: false,
        languages: data.discussionPreferences.languages };
    } }),
  withLoadingIndicator()
)(DumbExportSection);