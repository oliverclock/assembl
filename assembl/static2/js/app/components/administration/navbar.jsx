import React from 'react';
import { browserHistory } from 'react-router';
import { Translate } from 'react-redux-i18n';
import { Col, OverlayTrigger } from 'react-bootstrap';
import { getDiscussionSlug, calculatePercentage } from '../../utils/globalFunctions';
import { get } from '../../utils/routeMap';
import { nextStepTooltip, previousStepTooltip } from '../common/tooltips';

class Navbar extends React.Component {
  static defaultProps = {
    beforeChangeSection: () => {}
  };

  constructor(props) {
    super(props);
    const { currentStep, totalSteps } = this.props;
    this.state = {
      currentStep: currentStep,
      totalSteps: totalSteps
    };
    this.goToSection = this.goToSection.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      currentStep: nextProps.currentStep,
      totalSteps: nextProps.totalSteps
    });
  }

  goToSection(stepNb) {
    const slug = { slug: getDiscussionSlug() };
    const { phaseIdentifier } = this.props;
    browserHistory.push(
      `${get('administration', slug)}${get('adminPhase', { ...slug, phase: phaseIdentifier })}?section=${stepNb}`
    );
    this.setState({
      currentStep: stepNb
    });
  }

  render() {
    const { currentStep, totalSteps } = this.state;
    const { beforeChangeSection } = this.props;
    const barWidth = calculatePercentage(currentStep, totalSteps);
    return (
      <div className="admin-navbar">
        <Col xs={6} md={6}>
          <div className="step-numbers">
            <div className="txt">
              <Translate value="administration.step_x_total" num={currentStep} total={totalSteps} />
            </div>
            <div className="bar" style={{ width: `${barWidth}%` }}>
              &nbsp;
            </div>
            <div className="bkg-bar">&nbsp;</div>
          </div>
        </Col>
        <Col xs={6} md={6}>
          <div className="arrow-container">
            {currentStep < totalSteps && (
              <OverlayTrigger placement="top" overlay={nextStepTooltip}>
                <div
                  onClick={() => {
                    beforeChangeSection();
                    this.goToSection(currentStep + 1);
                  }}
                  className="arrow right"
                >
                  <span className="assembl-icon-up-open" />
                </div>
              </OverlayTrigger>
            )}
            {currentStep > 1 && (
              <OverlayTrigger placement="top" overlay={previousStepTooltip}>
                <div
                  onClick={() => {
                    beforeChangeSection();
                    this.goToSection(currentStep - 1);
                  }}
                  className="arrow right"
                >
                  <span className="assembl-icon-down-open" />
                </div>
              </OverlayTrigger>
            )}
          </div>
        </Col>
      </div>
    );
  }
}

export default Navbar;