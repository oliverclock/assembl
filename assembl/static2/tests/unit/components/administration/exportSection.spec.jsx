import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { DumbExportSection } from '../../../../js/app/components/administration/exportSection';

describe('ExportSection component', () => {
  it('should render an ExportSection component without languages options', () => {
    const props = {
      exporttype: 'voteSession',
      voteSessionId: '123'
    };
    const renderer = new ShallowRenderer();
    renderer.render(<DumbExportSection {...props} />);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
  it('should render an ExportSection component with languages options', () => {
    const props = {
      exporttype: 'survey',
      voteSessionId: '123',
      languages: [{ locale: 'fr', name: 'French', nativeName: 'fran√ßais', __typename: 'LocalePreference' },
        { locale: 'en', name: 'English', nativeName: 'English', __typename: 'LocalePreference' }
      ]
    };
    const renderer = new ShallowRenderer();
    renderer.render(<DumbExportSection {...props} />);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
});