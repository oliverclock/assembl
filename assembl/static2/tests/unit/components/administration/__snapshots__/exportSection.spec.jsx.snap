// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportSection component should render an ExportSection component with languages options 1`] = `null`;

exports[`ExportSection component should render an ExportSection component without languages options 1`] = `null`;
